state-chain/pallets/cf-swapping/src/tests/dca.rs

```diff
@@ -2,6 +2,100 @@ use super::*;
 
 const CHUNK_INTERVAL: u32 = 3;
 
+#[track_caller]
+fn setup_dca_swap(
+	number_of_chunks: u32,
+	chunk_interval: u32,
+	refund_params: Option<TestRefundParams>,
+) {
+	// Sanity check that the test started at the correct block
+	assert_eq!(System::block_number(), INIT_BLOCK);
+
+	// Start the dca swap
+	insert_swaps(&[TestSwapParams::new(
+		Some(DcaParameters { number_of_chunks, chunk_interval }),
+		refund_params,
+		false,
+	)]);
+
+	// Check that the swap request was received;
+	assert_has_matching_event!(
+		Test,
+		RuntimeEvent::Swapping(Event::SwapRequested {
+			swap_request_id: SWAP_REQUEST_ID,
+			input_amount: INPUT_AMOUNT,
```

Comment:
In this case, it looks like we just want to extract the input amount so we can calculate the chunk amount (the input amount doesn't matter: we want to know that the input, chunks and chunk amount are consistent).

Reply:
I think we want tests to highlight what's important (i.e. how a given test case is different from others) and the input amount, like you said, is not important here. If we need a specific/different amount for some reason, we can generalise this or simply write custom setup code. I don't think we need to generalise prematurely.

Reply:
I'm not talking about generalising for the sake of it. If the value is not important, so why are we checking it? For a reader of the tests, seeing this check suggests that the INPUT_AMOUNT is somehow relevant, and automatically then thinks, but why? And where is it defined? What if it's different? I think proper scoping, or using default test setups where it's clear that we're using, effectively, a placeholder value, would make this clearer.

Reply:
I'm not sure I understand your concerns. It is still important to use *some* value and check against it *and* other values derived from it. You can let rust-analyzer show where it is defined. It is defined externally, because other pieces of the same test rely on it. It is global so we don't have to pass it around, which make tests more clear. If it needs to be differnet for some reason in some test, then we can write a more specialized test just for that case, and we don't have to generalize everything just for that one case.

> it looks like we just want to extract the input amount so we can calculate the chunk amount

I see what you mean, but still, the same value is used (or derived from) in other places, so it is not good enough to check that something is "locally correct".